tests/components/deako/test_light.py

```diff
@@ -0,0 +1,275 @@
+"""Tests for the light module."""
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from homeassistant.components.deako.const import DOMAIN
+from homeassistant.components.deako.light import async_setup_entry
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    DOMAIN as LIGHT_DOMAIN,
+    ColorMode,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_OFF,
+    STATE_ON,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+
+from .conftest import MockDeakoDevices
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.usefixtures("init_integration")
+@pytest.mark.asyncio
+async def test_light_setup_no_devices(
+    hass: HomeAssistant,
+    pydeako_deako_mock: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test light platform setup with no devices returned."""
+
+    # no devices
+    pydeako_deako_mock.get_devices.return_value = {}
+
+    light_entities = []
+
+    async def mock_add_entities(entities):
+        for entity in entities:
+            light_entities.append(entity)
+
+    await async_setup_entry(hass, mock_config_entry, mock_add_entities)
```

Comment:
Idem. Just let the integration set itself up and test the state of the entities via the state machine